source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -495,6 +497,10 @@ bool CompressorFilter::isEtagAllowed(Http::ResponseHeaderMap& headers) const {
 
 bool CompressorFilterConfig::DirectionConfig::isMinimumContentLength(
     const Http::RequestOrResponseHeaderMap& headers) const {
+  if (StringUtil::caseFindToken(headers.getTransferEncodingValue(), ",",
```

Comment:
can this ever be true? I think the transfer encoding headers are removed before the filter ever sees it. It would also result in HTTP/1.1 and HTTP/2 responses without content length being treated differently.  I think this was buggy prior to your change but I'd be inclined to fix and test while we're in here.

Reply:
why h1 and h2 would be different? (i this diff we do not distinct h1 vs h2 in compressor. the runtime feature just either permit or deny compression if no content-length header). and yes it seems that you are right. at least manual testing shows that there is no compression if only trunsfer-encoding=chunked present. do you propose by somehow passing transfer-encoding into the filter or just remove this condition? I think originally idea was that is it is chunked - we making assumption that response is big enough and worth compressing (the whole idea behind isMinimumContentLength (from my understanding) is to be able to do some kind of performance optimization (do not compress something which is really small)) so from my point of view we can just omit this logic (unless @rojkov has some other thoughts on this)

Reply:
So HTTP/1 on the wire, has a transfer-encoding: chunked header for (most) responses without a content length.  The alternate is framing by connection close.
HTTP/2 disallows the transfer encoding header though data is implicitly chunked up by HTTP/2 data frames.

In Envoy, the HCM removes any transfer encoding header before this code is called.  If you add an integration test with T-E:chunked, and an assert here, I believe it can never trigger.
I believe the code needs to simply have one behavior if content length is present, and one if content length is absent, and not rely on the transfer encoding header.

Reply:
This code is quite old. I guess the original thinking was indeed to assume the payload is big enough if `transfer-encoding: chunked` is present.

> In Envoy, the HCM removes any transfer encoding header before this code is called.

Is it about the `chunked` part only or fully removed? If any t-e header is removed in HCM then also `CompressorFilter::isTransferEncodingAllowed()` becomes useless.

Reply:
Hm, I thought it was removed early but looks like I may be incorrect:

https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_utility.cc#L398
response_headers.removeTransferEncoding();

called here:
https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1385

which looks like it's right before it's sent the codec.  Either way a test would be good to make sure we're consistent between HTTP/1 and HTTP/2 without content length, SG?

Reply:
so my understanding here is that before any filters in http conn manager we are calling this: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1362
and especially this: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1385

which seems like a helper which always remove TE headers: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_utility.cc#L398

so it feels that it removes all TE. not just chunked. however i could miss some corner case as i'm not super familiar with conn_manager code path

Reply:
What tests are we talking about? for the WS/Update (to make sure that h1 and h2 are the same) or something else?